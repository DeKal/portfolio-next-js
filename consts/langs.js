export const EN = 'en'
export const VN = 'vn'
