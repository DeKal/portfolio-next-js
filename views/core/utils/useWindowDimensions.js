import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
