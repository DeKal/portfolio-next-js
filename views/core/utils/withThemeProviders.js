import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
