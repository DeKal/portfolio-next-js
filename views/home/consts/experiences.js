const row1 = ['Zalora', 'Dek']
const row2 = ['GCS', 'GCS-intern']
const row3 = ['SmartApp', 'Brand360']

export const getCompanies = () => [row1, row2, row3]
