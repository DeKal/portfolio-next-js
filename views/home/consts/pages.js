export const HOME = 'Home'
export const ABOUT = 'About'
export const EXPERIENCES = 'Experiences'
export const CERTIFICATIONS = 'Certifications'
export const PROJECTS = 'Projects'
export const CONTACTS = 'Contacts'
export const WHYME = 'Whyme'
export const BLOG = 'Blogs'

export const PAGE_LIST = [
  {
    name: HOME,
    url: `/#${HOME}`,
    cmpName: HOME
  },
  {
    name: ABOUT,
    url: `/#${ABOUT}`,
    cmpName: ABOUT
  },
  {
    name: EXPERIENCES,
    url: `/#${EXPERIENCES}`,
    cmpName: EXPERIENCES
  },
  {
    name: PROJECTS,
    url: `/#${PROJECTS}`,
    cmpName: PROJECTS
  },
  {
    name: CERTIFICATIONS,
    url: `/#${CERTIFICATIONS}`,
    cmpName: CERTIFICATIONS
  },
  {
    name: CONTACTS,
    url: `/#${CONTACTS}`,
    cmpName: CONTACTS
  },
  {
    name: BLOG,
    url: `/blogs`,
    cmpName: 'blogs'
  }
]
