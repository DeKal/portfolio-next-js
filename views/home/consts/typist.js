export const MAX_LIMIT = 999999999999
export const SEC_PER_TURN = 3
export const START_COUNT = 0
export const INTERVAL = 1000
