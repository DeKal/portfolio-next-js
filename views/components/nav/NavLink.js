import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { EN } from '~/consts/langs'
import { Translate, withLocalize } from 'react-localize-redux'

const NavLink = ({
  lang,
  selectedLang,
  setSelectedLang,
  setActiveLanguage
}) => {
  return (
    <Container
      isSelected={selectedLang === lang}
      onClick={() => {
        setSelectedLang(lang)
        setActiveLanguage(lang)
        window.localStorage.setItem('lang', lang)
      }}
    >
      {lang === EN ? <Translate id="english" /> : <Translate id="vietnamese" />}
    </Container>
  )
}

NavLink.propTypes = {
  lang: PropTypes.string,
  selectedLang: PropTypes.string,
  setSelectedLang: PropTypes.func,
  setActiveLanguage: PropTypes.func
}

export default withLocalize(NavLink)

const Container = styled.a`
  color: ${props =>
    props.isSelected
      ? props.theme.colors.colorDark
      : props.theme.colors.colorLight};
  border: ${props => (props.isSelected ? 'none' : 'initial')};
  border-color: ${props => props.theme.colors.colorLight};
  font-family: ${props => props.theme.font};
  font-style: italic;
  line-height: ${props => props.theme.navHeaderHeight};
  transition: color 0.5s;

  &:hover {
    color: ${props => props.theme.colors.colorDark};
    border: none;
  }
`
