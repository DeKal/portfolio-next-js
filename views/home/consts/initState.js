import { HOME } from '~/views/home/consts/pages'
import { EN } from '~/views/home/consts/langs'

export const INIT_NAV_CONTENT_SHOW = false
export const INIT_SELECTED_PAGE = HOME
export const INIT_SELECTED_LANG = EN
