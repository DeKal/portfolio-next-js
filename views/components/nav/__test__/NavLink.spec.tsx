import * as React from 'react'
import { mountWithTheme } from '~/utils/withThemeProviders'
import NavLink from '~/views/components/nav/NavLink'
import { EN, VN } from '~/consts/langs'

import useWindowDimensions from '~/utils/useWindowDimensions'
jest.mock('~/utils/useWindowDimensions')

describe('Nav Link', () => {
  describe('Big Device >= 600px', () => {
    beforeEach(() => {
      useWindowDimensions.mockReturnValue({
        width: 800
      })
    })
  
    it('should render with English selected', function () {
      const wrap = mountWithTheme(<NavLink lang={EN} selectedLang={EN} />)
      expect(wrap).toMatchSnapshot()
    })
  
    it('should render with Vietnamese selected', function () {
      const wrap = mountWithTheme(<NavLink lang={VN} selectedLang={VN} />)
      expect(wrap).toMatchSnapshot()
    })
  
    it('should change Language when clicking', function () {
      const setSelectedLang = jest.fn()
      const setActiveLanguage = jest.fn()
      const wrap = mountWithTheme(
        <NavLink
          lang={VN}
          selectedLang={EN}
          setSelectedLang={setSelectedLang}
          setActiveLanguage={setActiveLanguage}
        />
      )
      wrap.simulate('click')
      expect(wrap).toMatchSnapshot()
  
      expect(setSelectedLang.mock.calls.length).toEqual(1)
      expect(setSelectedLang).toHaveBeenCalledWith(VN)
  
      expect(setActiveLanguage.mock.calls.length).toEqual(1)
      expect(setActiveLanguage).toHaveBeenCalledWith(VN)
    })
  })
  
  describe('Small Device < 600px', () => {
    beforeEach(() => {
      useWindowDimensions.mockReturnValue({
        width: 500
      })
    })
  
    it('should render with English selected', function () {
      const wrap = mountWithTheme(<NavLink lang={EN} selectedLang={EN} />)
      expect(wrap).toMatchSnapshot()
    })
  
    it('should render with Vietnamese selected', function () {
      const wrap = mountWithTheme(<NavLink lang={VN} selectedLang={VN} />)
      expect(wrap).toMatchSnapshot()
    })
  })

})
