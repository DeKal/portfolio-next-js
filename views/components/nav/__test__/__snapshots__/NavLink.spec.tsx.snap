// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Link Big Device >= 600px should change Language when clicking 1`] = `
<LocalizedComponent
  lang="vn"
  selectedLang="en"
  setActiveLanguage={
    [MockFunction] {
      "calls": Array [
        Array [
          "vn",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  setSelectedLang={
    [MockFunction] {
      "calls": Array [
        Array [
          "vn",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="vn"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="en"
    setActiveLanguage={
      [MockFunction] {
        "calls": Array [
          Array [
            "vn",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    setSelectedLang={
      [MockFunction] {
        "calls": Array [
          Array [
            "vn",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={false}
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "NavLink__Container-v6tuz9-0",
              "isStatic": false,
              "lastClassName": "bfwuOx",
              "rules": Array [
                "color:",
                [Function],
                ";border:",
                [Function],
                ";border-color:",
                [Function],
                ";font-family:",
                [Function],
                ";font-style:italic;line-height:",
                [Function],
                ";transition:color 0.5s;&:hover{color:",
                [Function],
                ";border:none;}",
              ],
            },
            "displayName": "NavLink__Container",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "NavLink__Container-v6tuz9-0",
            "target": "a",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        isSelected={false}
        onClick={[Function]}
      >
        <a
          className="NavLink__Container-v6tuz9-0 bfwuOx"
          onClick={[Function]}
        >
          <Translate
            id="vietnamese"
          >
            <WrappedTranslate
              context={
                Object {
                  "activeLanguage": undefined,
                  "addTranslation": [Function],
                  "addTranslationForLanguage": [Function],
                  "defaultLanguage": undefined,
                  "ignoreTranslateChildren": false,
                  "initialize": [Function],
                  "languages": Array [],
                  "renderToStaticMarkup": false,
                  "setActiveLanguage": [Function],
                  "translate": [Function],
                }
              }
              id="vietnamese"
            >
              Missing translationId: vietnamese for language: \${ languageCode }
            </WrappedTranslate>
          </Translate>
        </a>
      </StyledComponent>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;

exports[`Nav Link Big Device >= 600px should render with English selected 1`] = `
<LocalizedComponent
  lang="en"
  selectedLang="en"
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="en"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="en"
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={true}
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "NavLink__Container-v6tuz9-0",
              "isStatic": false,
              "lastClassName": "fMGHjQ",
              "rules": Array [
                "color:",
                [Function],
                ";border:",
                [Function],
                ";border-color:",
                [Function],
                ";font-family:",
                [Function],
                ";font-style:italic;line-height:",
                [Function],
                ";transition:color 0.5s;&:hover{color:",
                [Function],
                ";border:none;}",
              ],
            },
            "displayName": "NavLink__Container",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "NavLink__Container-v6tuz9-0",
            "target": "a",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        isSelected={true}
        onClick={[Function]}
      >
        <a
          className="NavLink__Container-v6tuz9-0 fMGHjQ"
          onClick={[Function]}
        >
          <Translate
            id="english"
          >
            <WrappedTranslate
              context={
                Object {
                  "activeLanguage": undefined,
                  "addTranslation": [Function],
                  "addTranslationForLanguage": [Function],
                  "defaultLanguage": undefined,
                  "ignoreTranslateChildren": false,
                  "initialize": [Function],
                  "languages": Array [],
                  "renderToStaticMarkup": false,
                  "setActiveLanguage": [Function],
                  "translate": [Function],
                }
              }
              id="english"
            >
              Missing translationId: english for language: \${ languageCode }
            </WrappedTranslate>
          </Translate>
        </a>
      </StyledComponent>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;

exports[`Nav Link Big Device >= 600px should render with Vietnamese selected 1`] = `
<LocalizedComponent
  lang="vn"
  selectedLang="vn"
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="vn"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="vn"
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={true}
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "NavLink__Container-v6tuz9-0",
              "isStatic": false,
              "lastClassName": "fMGHjQ",
              "rules": Array [
                "color:",
                [Function],
                ";border:",
                [Function],
                ";border-color:",
                [Function],
                ";font-family:",
                [Function],
                ";font-style:italic;line-height:",
                [Function],
                ";transition:color 0.5s;&:hover{color:",
                [Function],
                ";border:none;}",
              ],
            },
            "displayName": "NavLink__Container",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "NavLink__Container-v6tuz9-0",
            "target": "a",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        isSelected={true}
        onClick={[Function]}
      >
        <a
          className="NavLink__Container-v6tuz9-0 fMGHjQ"
          onClick={[Function]}
        >
          <Translate
            id="vietnamese"
          >
            <WrappedTranslate
              context={
                Object {
                  "activeLanguage": undefined,
                  "addTranslation": [Function],
                  "addTranslationForLanguage": [Function],
                  "defaultLanguage": undefined,
                  "ignoreTranslateChildren": false,
                  "initialize": [Function],
                  "languages": Array [],
                  "renderToStaticMarkup": false,
                  "setActiveLanguage": [Function],
                  "translate": [Function],
                }
              }
              id="vietnamese"
            >
              Missing translationId: vietnamese for language: \${ languageCode }
            </WrappedTranslate>
          </Translate>
        </a>
      </StyledComponent>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;

exports[`Nav Link Small Device < 600px should render with English selected 1`] = `
<LocalizedComponent
  lang="en"
  selectedLang="en"
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="en"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="en"
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={true}
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "NavLink__Container-v6tuz9-0",
              "isStatic": false,
              "lastClassName": "fMGHjQ",
              "rules": Array [
                "color:",
                [Function],
                ";border:",
                [Function],
                ";border-color:",
                [Function],
                ";font-family:",
                [Function],
                ";font-style:italic;line-height:",
                [Function],
                ";transition:color 0.5s;&:hover{color:",
                [Function],
                ";border:none;}",
              ],
            },
            "displayName": "NavLink__Container",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "NavLink__Container-v6tuz9-0",
            "target": "a",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        isSelected={true}
        onClick={[Function]}
      >
        <a
          className="NavLink__Container-v6tuz9-0 fMGHjQ"
          onClick={[Function]}
        >
          en
        </a>
      </StyledComponent>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;

exports[`Nav Link Small Device < 600px should render with Vietnamese selected 1`] = `
<LocalizedComponent
  lang="vn"
  selectedLang="vn"
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="vn"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="vn"
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={true}
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "NavLink__Container-v6tuz9-0",
              "isStatic": false,
              "lastClassName": "fMGHjQ",
              "rules": Array [
                "color:",
                [Function],
                ";border:",
                [Function],
                ";border-color:",
                [Function],
                ";font-family:",
                [Function],
                ";font-style:italic;line-height:",
                [Function],
                ";transition:color 0.5s;&:hover{color:",
                [Function],
                ";border:none;}",
              ],
            },
            "displayName": "NavLink__Container",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "NavLink__Container-v6tuz9-0",
            "target": "a",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        isSelected={true}
        onClick={[Function]}
      >
        <a
          className="NavLink__Container-v6tuz9-0 fMGHjQ"
          onClick={[Function]}
        >
          vn
        </a>
      </StyledComponent>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;
